browser/android/brave_shields_content_settings.cc

```diff
@@ -0,0 +1,202 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/android/brave_shields_content_settings.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+
+#include <string>
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "components/content_settings/core/browser/content_settings_utils.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+
+#include "jni/BraveShieldsContentSettings_jni.h"
+
+
+namespace chrome {
+namespace android {
+
+BraveShieldsContentSettings* g_brave_shields_content_settings = nullptr;
```

Comment:
Can we add a comment here explaining that this is linked to the global toolbar in java? Otherwise you have to track through a lot of code to understand how this can be a single global instance

Reply:
Done